### PR TITLE
feat: FE lib

### LINKED ISSUE #517: shared api
develop a shared TS api for all apps, web, mobile, etc

### PR DESCRIPTION
**Pull request by garrettladley**
@DOOduneye can you take the auth from here?

### CODE REVIEW

File: frontend/lib/api/categories.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { z } from "zod";
```
Comment (alderwhiteford) [inline frontend/lib/api/categories.ts:1]: more of a general comment. Seeing that we are still using zustand for state management for auth. Do we want to migrate this to redux so it is more compatible with RTK?

Comment (alderwhiteford) [reply frontend/lib/api/categories.ts:1]: Realizing this might be something you're still working ^^

File: frontend/lib/api/club.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  application_link: z.string().max(255),
+  logo: z.string().max(255).optional(),
+});
+
+export const clubSchema = clubSchemaIntermediate.merge(rootModelSchema);
+export type Club = z.infer<typeof clubSchema>;
+
+export const clubsApi = createApi({
+  reducerPath: "clubsApi",
+  baseQuery: fetchBaseQuery({ baseUrl: API_BASE_URL }),
```
Comment (alderwhiteford) [inline frontend/lib/api/club.ts:26]: Do we need to pass any headers to our requests? If so, we can use the prepareHeaders option

File: frontend/lib/api/categories.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import { tagSchema } from "./tag";
+
+export const categorySchemaIntermediate = z.object({
+  name: z.string(),
+  tags: z.array(tagSchema),
+});
+
+export const categorySchema = categorySchemaIntermediate.merge(rootModelSchema);
+export type Category = z.infer<typeof categorySchema>;
```
Comment (alderwhiteford) [inline frontend/lib/api/categories.ts:13]: Are we expecting to have devs call schema.parse when they fetch results? 

We might be able to take advantage of transformResponse to have this handled in the API itself. Might be worth looking into

Comment (alderwhiteford) [reply frontend/lib/api/categories.ts:13]: Really like the use of zod here

File: frontend/lib/api/club.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  endpoints: (builder) => ({
+    clubByID: builder.query<Club, uuid>({
+      query: (clubID) => `clubs/${clubID}`,
+    }),
+    clubContactsByClubID: builder.query<Contact[], uuid>({
```
Comment (alderwhiteford) [inline frontend/lib/api/club.ts:31]: Do we have any non get requests? Seeing everything is a query

File: frontend/mobile/components/club-contacts.tsx [hunk 5]
```diff
@@ -1,20 +1,25 @@
...
 
 import { ContactIcon } from './contact-icon';
 
 const ClubContacts = ({ clubID }: { clubID: uuid }) => {
-    const { data: clubContacts, error, isLoading } = useClubContacts(clubID);
+    const {
+        data: clubContacts,
+        error,
+        isLoading
+    } = useClubContactsByClubIDQuery(clubID);
```
Comment (alderwhiteford) [inline frontend/mobile/components/club-contacts.tsx:14]: if transformResponse doesn't work for validation, we could also use: useClubContactByClubIDLazyQuery (or something like then). Gives us access to the .then .catch pattern

**Comment by garrettladley** [comment]
going to close for right now to avoid pre showcase grind into merge conflicts

**Pull request closed without merging**